django/template/base.py

```diff
@@ -199,6 +202,7 @@ def _render(self, context):
         return self.nodelist.render(context)
 
     def render(self, context):
+        context.template_name = self.name
```

Comment:
Please move this to below of the docstring.

```py
def render(self, context):
        "Display stage -- can be called many times"
        # Set engine attribute here to avoid changing the signature of either
        # Context.__init__ or Node.render. The engine is set only on the first
        # call to render. Further calls e.g. for includes don't override it.
        context.template_name = self.name
        # ...
```